fetch/fetch.go

```diff
@@ -157,45 +207,67 @@ type Fetch struct {
 	hashToPeers     *HashPeersCache
 }
 
-// newFetch creates a new Fetch struct.
-func newFetch(cfg Config, h host, bs *datastore.BlobStore, atxS, lyrS, opnS, hashS server.Requestor, logger log.Log) *Fetch {
+// NewFetch creates a new Fetch struct.
+func NewFetch(cdb *datastore.CachedDB, msh meshProvider, host *p2p.Host, opts ...Option) *Fetch {
+	bs := datastore.NewBlobStore(cdb.Database)
 	f := &Fetch{
-		cfg:             cfg,
-		log:             logger,
+		cfg:             DefaultConfig(),
+		logger:          log.NewNop(),
 		bs:              bs,
-		host:            h,
-		atxSrv:          atxS,
-		lyrSrv:          lyrS,
-		opnSrv:          opnS,
-		hashSrv:         hashS,
+		host:            host,
+		servers:         map[string]requester{},
 		activeRequests:  make(map[types.Hash32][]*request),
 		pendingRequests: make(map[types.Hash32][]*request),
 		requestReceiver: make(chan request),
-		batchTimeout:    time.NewTicker(time.Millisecond * time.Duration(cfg.BatchTimeout)),
 		stop:            make(chan struct{}),
 		activeBatches:   make(map[types.Hash32]batchInfo),
 		hashToPeers:     NewHashPeersCache(cacheSize),
 	}
+	for _, opt := range opts {
+		opt(f)
+	}
+
+	f.batchTimeout = time.NewTicker(time.Millisecond * time.Duration(f.cfg.BatchTimeout))
+	if len(f.servers) == 0 {
+		h := newHandler(cdb, bs, msh, f.logger)
+		timeout := time.Second * time.Duration(f.cfg.RequestTimeout)
```

Comment:
Would it be possible to make `RequestTimeout` a `time.Duration` instead of an `int`?